x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_panel.tsx

```diff
@@ -227,14 +227,57 @@ export const IndexPatternDimensionTriggerComponent = function IndexPatternDimens
   props: IndexPatternDimensionTriggerProps
 ) {
   const layerId = props.layerId;
-
-  const selectedColumn: IndexPatternColumn | null =
-    props.state.layers[layerId].columns[props.columnId] || null;
+  const layer = props.state.layers[layerId];
+  const selectedColumn: IndexPatternColumn | null = layer.columns[props.columnId] || null;
+  const currentIndexPattern = props.state.indexPatterns[layer.indexPatternId];
+
+  const currentFieldIsInvalid = useMemo(
+    () =>
+      fieldIsInvalid(
+        selectedColumn?.sourceField,
+        selectedColumn?.operationType,
+        currentIndexPattern
+      ),
+    [selectedColumn?.sourceField, selectedColumn?.operationType, currentIndexPattern]
+  );
 
   const { columnId, uniqueLabel } = props;
   if (!selectedColumn) {
     return null;
   }
+
+  if (currentFieldIsInvalid) {
```

Comment:
2. To change the styles to hover the whole element, you could add a class ` anchorClassName="lnsLayerPanel__anchor"` to tooltip. To make it prettier, I copied the elements from how we did filters and range elements:
```
  if (currentFieldIsInvalid) {
    return (
      <EuiToolTip
        content={i18n.translate('xpack.lens.configure.invalidConfigTooltip', {
          defaultMessage: "The field {field} can't be used. Please adjust the configuration.",
          values: {
            field: selectedColumn.sourceField,
          },
        })}
        anchorClassName="lnsLayerPanel__anchor"
      >
        <EuiLink
          color="danger"
          id={columnId}
          className="lnsLayerPanel__triggerLink"
          onClick={props.onClick}
          data-test-subj="lns-dimensionTrigger"
          aria-label={i18n.translate('xpack.lens.configure.editConfig', {
            defaultMessage: 'Edit configuration',
          })}
          title={i18n.translate('xpack.lens.configure.editConfig', {
            defaultMessage: 'Edit configuration',
          })}
        >
          <EuiFlexGroup gutterSize="s" alignItems="center" responsive={false}>
            <EuiFlexItem grow={false}>
              <EuiIcon size="s" type="alert" />
            </EuiFlexItem>
            <EuiFlexItem grow={true}>
              {i18n.translate('xpack.lens.configure.invalidConfig', {
                defaultMessage: 'Configuration is invalid',
              })}
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiLink>
      </EuiToolTip>
    );
  }
```

![image](https://user-images.githubusercontent.com/4283304/94433038-fbbba400-0197-11eb-8454-6cec9c9f3ff1.png)